src/qml/qgismobileapp.qml

```diff
@@ -279,27 +279,44 @@ ApplicationWindow {
       anchors.fill: parent
 
       onClicked:  {
-        if (locatorItem.state == "on") {
-          locatorItem.state = "off"
-        }
-        else if (geometryEditorsToolbar.canvasClicked(point)) {
-          // for instance, the vertex editor will select a vertex if possible
-        }
-        else if ( type === "stylus" && ( ( stateMachine.state === "digitize" && dashBoard.currentLayer ) || stateMachine.state === 'measure' ) ) {
-          if ( Number( currentRubberband.model.geometryType ) === QgsWkbTypes.PointGeometry ||
-              Number( currentRubberband.model.geometryType ) === QgsWkbTypes.NullGeometry )
+          // Check if geometry editor is taking over
+          if ( geometryEditorsToolbar.canvasClicked(point) )
+              return;
+
+          if (locatorItem.state == "on")
```

Comment:
Is this correct, the locator should stay open if the geometry editor is taking over?

Reply:
No, but the locator shouldn't be open if the geometry editor is on (which is a statement that isn't valid ATM, we can fix in a different PR).